flax/linen/module.py

```diff
@@ -133,9 +137,30 @@ def get_suffix_value_pairs(
     flat_dict = traverse_util.flatten_dict(dict_or_leaf)
     return [('_' + '_'.join(k), v) for k, v in flat_dict.items()]
 
-
-def all_names_on_object(obj: Any) -> Set[str]:
-  """Get all names of attributes on self and its classes throughout MRO."""
+def _is_module_tree(in_tree: Any) -> bool:
```

Comment:
It's a little worrying that this function is here at all - it was deleted a while ago during a simplification / fix.  Are you sure you based this commenting PR on a recent commit of flax?

Reply:
NIce catch! :-) I started this PR with a wrong base and we then rebased it, but I guess we forgot to remove this function. I removed it now.